src/lib.rs

```diff
@@ -28,12 +28,13 @@ use holochain_types::prelude::{AppManifest, MembraneProof, UnsafeBytes};
 
 pub async fn activate_holo_hosted_happs(core_happ: &Happ, config: &Config) -> Result<()> {
     let list_of_happs = get_all_enabled_hosted_happs(core_happ).await?;
-    install_holo_hosted_happs(list_of_happs, config).await?;
+    install_holo_hosted_happs(&list_of_happs, config).await?;
+    check_for_happs_to_be_uninstalled(&list_of_happs, config).await?;
```

Comment:
I'm not seeing how a happ is ever going to get uninstalled. Is the idea here that if a publisher calls `remove_happ`, then it will no longer be included in the return value of `get_all_enabled_hosted_happs? I don't think that's true, because all `remove_happ` does is delete an entry, but `get_happs` in HHA just checks a list of links. (So the links will still point to the deleted entry)

Reply:
Good catch. I think instead of deleting the entry I should just delete the links. What do you think? Or I can delete both

Reply:
I wrote tests for this in HHA, and `get_links` do not return links pointing to deleted entries.